src/components/pages/deals/Deals.js

```diff
@@ -20,14 +34,83 @@ const Deals = () => {
         fetch_deals()
     }, [])
 
-    const deals_to_display = isLoading === false && deals.length === 0 ? <div className="noDeals"><h1>No Deals available at the moment</h1></div> : <ProductsOnDeals items={deals}/>
+    // this function recieves the search text from the searchbar component and 
+    // sets it to the filter state and filters out some items depending on the text
+    const handleSearchedItems = (filter) => {
+
+        setSearchFilter(filter)
+
+        // this const is filtered items
+        const itemsSearched = deals.filter(deal => {
```

Comment:
You should rerun the item query with the search input in the `q` query param to search items via the API